star_tides/core/actions/base_action.py

```diff
@@ -0,0 +1,10 @@
+from abc import ABCMeta, abstractmethod
+
+
+class Action(metaclass=ABCMeta):
```

Comment:
Can you explain what all of these abstract classes are for? Or better yet, write docs?